assets/scripts/check-deeplink.js

```diff
@@ -0,0 +1,145 @@
+function _registerEvent(target, eventType, cb) {
+  if (target.addEventListener) {
+    target.addEventListener(eventType, cb);
+    return {
+      remove: function() {
+        target.removeEventListener(eventType, cb);
+      }
+    };
+  } else {
+    target.attachEvent(eventType, cb);
+    return {
+      remove: function() {
+        target.detachEvent(eventType, cb);
+      }
+    };
+  }
+}
+
+function _createHiddenIframe(target, uri) {
+  var iframe = document.createElement("iframe");
+  iframe.src = uri;
+  iframe.id = "hiddenIframe";
+  iframe.style.display = "none";
+  target.appendChild(iframe);
+
+  return iframe;
+}
+
+function openUriWithHiddenFrame(uri, failCb, successCb) {
+  var timeout = setTimeout(function() {
+    failCb();
+    handler.remove();
+  }, 1000);
+
+  var iframe = document.querySelector("#hiddenIframe");
+  if (!iframe) {
+    iframe = _createHiddenIframe(document.body, "about:blank");
+  }
+
+  var handler = _registerEvent(window, "blur", onBlur);
+
+  function onBlur() {
+    clearTimeout(timeout);
+    handler.remove();
+    successCb();
+  }
+
+  iframe.contentWindow.location.href = uri;
+}
+
+function openUriWithTimeoutHack(uri, failCb, successCb) {
+  var timeout = setTimeout(function() {
+    failCb();
+    handler.remove();
+  }, 1000);
+
+  //handle page running in an iframe (blur must be registered with top level window)
+  var target = window;
+  while (target != target.parent) {
+    target = target.parent;
+  }
+
+  var handler = _registerEvent(target, "blur", onBlur);
+
+  function onBlur() {
+    clearTimeout(timeout);
+    handler.remove();
+    successCb();
+  }
+
+  window.location = uri;
+}
+
+function openUriUsingFirefox(uri, failCb, successCb) {
+  var iframe = document.querySelector("#hiddenIframe");
+  if (!iframe) {
+    iframe = _createHiddenIframe(document.body, "about:blank");
+  }
+  try {
+    iframe.contentWindow.location.href = uri;
+    successCb();
+  } catch (e) {
+    if (e.name == "NS_ERROR_UNKNOWN_PROTOCOL") {
+      failCb();
+    }
+  }
+}
+
+function openUriWithMsLaunchUri(uri, failCb, successCb) {
+  navigator.msLaunchUri(uri, successCb, failCb);
```

Comment:
I didn't test the code on MS Edge, needs to be done during the validation of this PR.